src/node_sockaddr.h

```diff
@@ -116,6 +120,89 @@ class SocketAddress : public MemoryRetainer {
   sockaddr_storage address_;
 };
 
+// A BlockList is used to evaluate whether a given
+// SocketAddress should be accepted for inbound or
+// outbound network activity.
+class SocketAddressBlockList {
+ public:
+  SocketAddressBlockList() = default;
+  ~SocketAddressBlockList() = default;
+
+  void AddSocketAddress(
+      const SocketAddress& address);
+
+  void AddSocketAddressRange(
+      const SocketAddress& start,
+      const SocketAddress& end);
+
+  void AddSocketAddressMask(
+      const SocketAddress& address,
+      int prefix);
+
+  bool Apply(const SocketAddress& address);
+
+  size_t size() const { return rules_.size(); }
+
+  struct Rule {
+    virtual bool Apply(const SocketAddress& address) = 0;
+  };
+
+  struct SocketAddressRule : Rule {
+    SocketAddress address;
+
+    SocketAddressRule(const SocketAddress& address);
+
+    bool Apply(const SocketAddress& address) override;
+  };
+
+  struct SocketAddressRangeRule : Rule {
+    SocketAddress start;
+    SocketAddress end;
+
+    SocketAddressRangeRule(
+        const SocketAddress& start,
+        const SocketAddress& end);
+
+    bool Apply(const SocketAddress& address) override;
+  };
+
+  struct SocketAddressMaskRule : Rule {
+    SocketAddress address;
+    int prefix;
+
+    SocketAddressMaskRule(
+        const SocketAddress& address,
+        int prefix);
+
+    bool Apply(const SocketAddress& address) override;
+  };
+
+ private:
+  std::list<std::unique_ptr<Rule>> rules_;
+};
+
+class SocketAddressBlockListWrap :
+    public BaseObject,
+    public SocketAddressBlockList {
+ public:
+  static void Initialize(v8::Local<v8::Object> target,
+                         v8::Local<v8::Value> unused,
+                         v8::Local<v8::Context> context,
+                         void* priv);
+
+  static void New(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void AddAddress(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void Check(const v8::FunctionCallbackInfo<v8::Value>& args);
+
+  SocketAddressBlockListWrap(
+      Environment* env,
+      v8::Local<v8::Object> wrap);
+
+  SET_NO_MEMORY_INFO()
```

Comment:
This should report memory usage

Reply:
Oh yeah, it will. As I mentioned, implementation is still in progress ;-)